plugin/formatting.py

```diff
@@ -100,16 +102,70 @@ class LspFormatDocumentCommand(LspTextCommand):
 
     capability = 'documentFormattingProvider'
 
-    def is_enabled(self, event: Optional[dict] = None, point: Optional[int] = None) -> bool:
+    def is_enabled(self, event: Optional[dict] = None, select: bool = False) -> bool:
+        if select:
+            return len(list(self.sessions(self.capability))) > 1
         return super().is_enabled() or bool(self.best_session(LspFormatDocumentRangeCommand.capability))
 
-    def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
-        format_document(self).then(self.on_result)
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None, select: bool = False) -> None:
+        session_names = [session.config.name for session in self.sessions(self.capability)]
+        base_scope = self.view.scope_name(0).split()[0]
+        if select:
+            self.select_formatter(base_scope, session_names)
+        elif len(session_names) > 1:
+            window = self.view.window()
+            if not window:
+                return
+            formatter = None
+            project_data = window.project_data()
+            if isinstance(project_data, dict):
+                formatter = project_data.get('settings', {}).get('LSP', {}).get('formatters', {}).get(base_scope)
```

Comment:
We could instead do:

```suggestion
                formatter = DottedDict(project_data).get('settings.LSP.formatters.{}'.format(base_scope))
```